### PR TITLE
Handle raw data generation

### PR DESCRIPTION
**Pull request by zoontek**
As Babel community starts talking about an autoprefixer-like preset, we need to extract the raw data in other format that HTML.

It's a first and basic step, maybe we should split in several repos someday (?)
- compat-table/generation (containing JavaScript tests)
- compat-table/raw-data (JSON generation using compat-table/generation)
- compat-table/website (HTML generation using compat-table/generation)

It's just an idea of course.

Refs:
https://github.com/kangax/compat-table/issues/785
https://github.com/babel/babel/pull/3476

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -356,24 +356,42 @@ function handle(options) {
...
+  var html_result = replaceAndIndent(html, [
     ["<!-- NAME -->", [options.name]],
     ["<!-- URL -->", [options.name.link(options.url)]],
     ["<!-- POLYFILLS -->", !options.polyfills ? [] : options.polyfills.map(function(e) {
       return '<script>' + fs.readFileSync(__dirname + path.sep + e, 'utf-8').replace(/<(?=\/script>)/g,'\\u003c') + '</script>\n';
     })],
   ]).replace(/\t/g, '  ');
+  var json_result = JSON.stringify(options.tests, null, 2);
+
+  var html_target_file = __dirname + path.sep + options.html_target_file;
```
Comment (ljharb) [inline build.js]: Let's use path.join here, instead of having to manually use path.sep

**Comment by ljharb** [comment]
I like this approach, but our data has functions in it, and JSON does not support functions.

**Comment by zoontek** [comment]
@ljharb Are data functions the ones used to generate the "Current browser" column?

**Comment by chicoxyzzy** [comment]
@ljharb why should browser data include functions? Most likely we can replace them by strings or numbers.

**Comment by ljharb** [comment]
@chicoxyzzy for the raw list of browsers, JSON is totally fine. I'm talking about the actual tests, which require the code to use for the test, which are functions. Converting them to strings isn't the right direction imo, as already they rely too heavily on `eval`.

**Comment by chicoxyzzy** [comment]
@zoontek indeed. your PR is missing browser tests. Of corse this repo can be splitted some day but I think we should move step by step with fully functioning intermediate changes.

**Comment by chicoxyzzy** [comment]
Another thoughts for future steps:
- browser lists should be merged for all browsers
- browser lists shouldn't include browser ranges
- browsers with higher versions shouldn't nest data from previous versions implicitly

**Comment by chicoxyzzy** [comment]
Revisiting this PR.

I think it's good but we should add fields which link test to actual test body (or even better to tests in https://github.com/tc39/test262 in the future)

**Comment by chicoxyzzy** [comment]
ref: https://github.com/kangax/compat-table/issues/830

**Comment by chicoxyzzy** [comment]
This PR is outdated. I'm going to solve this problem in another way. Thank you for working on this!

**Pull request closed without merging**